server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheRequestProcessor.java

```diff
@@ -234,8 +234,14 @@ void put(HotRodHeader header, Subject subject, byte[] key, byte[] value, Metadat
 
    private void putInternal(HotRodHeader header, AdvancedCache<byte[], byte[]> cache, byte[] key, byte[] value,
                             Metadata metadata, InfinispanSpan<CacheEntry<?, ?>> span) {
-      cache.putAsyncEntry(key, value, metadata)
-            .whenComplete((ce, throwable) -> handlePut(header, ce, throwable))
+      CompletionStage<CacheEntry<byte[], byte[]>> cs;
+      if (header.hasFlag(ProtocolFlag.ForceReturnPreviousValue)) {
+         cs = cache.putAsyncEntry(key, value, metadata);
+      } else {
+         cs = cache.putAsync(key, value, metadata)
```

Comment:
This won't do much of anything. You want to invoke `cache.withFlags(Flag.IGNORE_RETURN_VALUES)`.